utils/exceptions.hh

```diff
@@ -45,3 +48,10 @@ public:
 
     const std::error_code& code() const { return _code; }
 };
+
+// Controls whether on_internal_error() aborts or throws.
+void set_abort_on_internal_error(bool do_abort);
+
+// Handles reporting of violation of internal invariants.
+// Callers can assume that it does not return. May throw.
+void on_internal_error(seastar::logger&, const seastar::sstring& reason);
```

Comment:
We actually only need to pass a `const char*` rather than sstring& this will reduce header dependencies a bit.

Reply:
Ok.